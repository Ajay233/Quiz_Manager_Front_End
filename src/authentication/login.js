import React from 'react';
import { connect } from 'react-redux';
import Notification from '../notifications/notifications'
import LoginForm from './forms/loginForm'

import { setUser, setRedirect } from './actions'
import { setNotification } from '../notifications/actions';

import '../stylesheets/login.css'

export class Login extends React.Component {

  componentDidMount(){
    document.documentElement.scrollTop = 0;
  }

  // AFTER back end update on login endpoint:
  // update action creator so it sets the correct error notification for incorrect credentials or
  // user has not yet verified their email address

  render(){
    const { setUser } = this.props
    return(
      <div className="loginContainer">
        <div className="login">
        <div className="notificationContainer">
          <Notification />
        </div>
          <LoginForm setUser={setUser} />
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    userData: state.userData,
    verificationProcess: state.verificationProcess
  };
}

export default connect(mapStateToProps, {setUser, setRedirect, setNotification})(Login);
