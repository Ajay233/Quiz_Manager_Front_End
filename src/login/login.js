import React from 'react';
import { Link } from 'react-router-dom';

class Login extends React.Component {
  render(){
    return(
      <div>
        Login below
        <form>
          <input></input>
          <input></input>
          <button>Go</button>
        </form>
        <div>
          Don't have an account? <Link to="/signUp">Sign up</Link> here
        </div>
      </div>
    );
  }
}

export default Login;
