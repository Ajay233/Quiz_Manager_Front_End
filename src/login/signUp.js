import React from 'react';

class SignUp extends React.Component {
  render(){
    return(
      <div>
        Sign up below
        <form>
          <input></input>
          <input></input>
          <input></input>
          <input></input>
          <button>Go</button>
        </form>
      </div>
    );
  }
}

export default SignUp;
