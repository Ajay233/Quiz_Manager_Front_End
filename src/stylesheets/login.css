.loginContainer {
  background-image: url('../public/backgrounds/login.jpg');
  background-size: cover;
  height: 100%;
}

.login {
  position: absolute;
  font-family: arial;
  color: white;
  background-color: rgba(30, 30, 30, 1);
  width: 35%;
  height: 100%;
  z-index: 0;
  color: white;
}

.loginForm {
  width: 80%;
  margin: 25% 10% 5% 10%;
}

.loginTitle {
  font-size: 40px;
  font-weight: bold;
  color: white;
  margin-bottom: 30px;
}

#signUpLink {
  text-align: center;
}

.loginButton {
  width: 40%;
  margin-left: 30%;
  margin-right: 30%;
}

.notificationContainer{
  height: 60px;
  margin-top: 50px;
}
