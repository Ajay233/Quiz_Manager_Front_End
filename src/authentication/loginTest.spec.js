import React from 'react'
import { Provider } from 'react-redux'
import { Router } from 'react-router-dom'
import { Login, mapStateToProps } from './login'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import history from '../history'

const mockStore = configureStore({})

describe("mapStateToProps", () => {
  it("should map state to props", () => {
    const appState = {
      userData: { id: 1, jwt: "jwt" },
      verificationProcess: { completionStatus: "test", token: "test", error: {} }
    }

    const componentState = mapStateToProps(appState)
    expect(componentState).toEqual(appState)
  })
})

describe("Login", () => {
  it("should render the login form without notification", () => {

    const store = mockStore({
      userData: { id: 1, jwt: "jwt" },
      verificationProcess: { completionStatus: "test", token: "test", error: {} },
      notificationData: { message: "", type: "", show: false, timed: true },
      globals: { showLoader: false }
    })

    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Login globals={{ showLoader: false }}/>
        </Router>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  })

  it("should render the login form with notification", () => {

    const store = mockStore({
      userData: { id: 1, jwt: "jwt" },
      verificationProcess: { completionStatus: "test", token: "test", error: {} },
      notificationData: { message: "Username or passord incorrect", type: "error", show: true, timed: true },
      globals: { showLoader: false }
    })

    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Login globals={{ showLoader: false }}/>
        </Router>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  })
})
