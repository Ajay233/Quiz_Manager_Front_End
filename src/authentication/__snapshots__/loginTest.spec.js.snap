// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the login form with notification 1`] = `
<div
  className="loginContainer"
>
  <div
    className="login"
  >
    <div
      className="notificationContainer"
    >
      <div
        className="spacing error"
        id="notification"
      >
        Username or passord incorrect
      </div>
    </div>
    <div>
      <form
        className="loginForm"
        onSubmit={[Function]}
      >
        <div
          className="loginTitle"
        >
          Login
        </div>
        <div>
          <label>
            Username (Email address):
          </label>
          <input
            className="inputBox"
            name="userName"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type=""
            value=""
          />
        </div>
        <div>
          <label>
            Password:
          </label>
          <input
            className="inputBox"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="password"
            value=""
          />
        </div>
        <button
          className="submit loginButton"
        >
          Login
        </button>
        <hr />
        <div
          id="signUpLink"
        >
          Don't have an account? 
          <a
            href="/signUp"
            onClick={[Function]}
          >
            Sign up
          </a>
           here
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Login should render the login form without notification 1`] = `
<div
  className="loginContainer"
>
  <div
    className="login"
  >
    <div
      className="notificationContainer"
    />
    <div>
      <form
        className="loginForm"
        onSubmit={[Function]}
      >
        <div
          className="loginTitle"
        >
          Login
        </div>
        <div>
          <label>
            Username (Email address):
          </label>
          <input
            className="inputBox"
            name="userName"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type=""
            value=""
          />
        </div>
        <div>
          <label>
            Password:
          </label>
          <input
            className="inputBox"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="password"
            value=""
          />
        </div>
        <button
          className="submit loginButton"
        >
          Login
        </button>
        <hr />
        <div
          id="signUpLink"
        >
          Don't have an account? 
          <a
            href="/signUp"
            onClick={[Function]}
          >
            Sign up
          </a>
           here
        </div>
      </form>
    </div>
  </div>
</div>
`;
